parsons/action_kit/action_kit.py

```diff
@@ -54,21 +54,43 @@ def _base_get(self, endpoint, entity_id=None, exception_message=None):
         resp = self.conn.get(self._base_endpoint(endpoint, entity_id))
 
         if exception_message and resp.status_code == 404:
-            raise Exception(exception_message)
+            raise Exception(self.parse_error(resp, exception_message))
 
         return resp.json()
 
-    def _base_post(self, endpoint, exception_message, **kwargs):
+    def _base_post(self, endpoint, exception_message, return_full_json=False, **kwargs):
         # Make a general post request to ActionKit
 
         resp = self.conn.post(self._base_endpoint(endpoint), data=json.dumps(kwargs))
 
         if resp.status_code != 201:
-            raise Exception(exception_message)
+            raise Exception(self.parse_error(resp, exception_message))
 
-        if 'headers' in resp.__dict__:
+        # Some of the methods should just return pointer to location of created
+        # object.
+        if 'headers' in resp.__dict__ and not return_full_json:
             return resp.__dict__['headers']['Location']
 
+        # Not all responses return a json
+        try:
+            return resp.json()
+
+        except ValueError:
+            return None
+
+    def parse_error(self, resp, exception_message):
+        # AK provides some pretty robust/helpful error reporting. We should surface them with
+        # our exceptions.
```

Comment:
It's probably worth checking to see if the response is json and throwing a generic "unexpected response" error if it's not.